go/libraries/doltcore/sqle/dtables/unscoped_diff_table.go

```diff
@@ -189,39 +202,158 @@ func (itr *UnscopedDiffTableItr) calculateChangedTables(commit *doltdb.Commit) (
 		return nil, err
 	}
 
-	tablesMap := make(map[string]bool)
+	tableChanges := make([]tableChange, 0)
```

Comment:
When you know the length of the slice ahead of time, it's more idiomatic to declare it here (`len(deltas)`) and then do direct assignment in the loop (`tableChanges[i] = ...`) rather than use append. This avoids the cost of growing the slice and backing array several times.

Also with golang range loops, if you're using `i` you just want to use the simpler form and not declare `delta` as well.